test/src/test/java/org/corfudb/recovery/FastObjectLoaderTest.java

```diff
@@ -813,233 +472,134 @@ public void canReadWithEntriesInterleavingCPS() throws Exception {
+        assertThatMapsAreBuilt(rt2);
+        assertThatObjectCacheIsTheSameSize(getDefaultRuntime(), rt2);
 
     }
 
     @Test
     public void canReadLogTerminatedWithCheckpoint() throws Exception{
-        CorfuRuntime rt1 = getDefaultRuntime();
-        Map<String, String> map1 = createMap("Map1", rt1);
-        Map<String, String> map2 = createMap("Map2", rt1);
-        Map<String, String> map3 = createMap("Map3", rt1);
+        populateMaps(SOME, getDefaultRuntime(), CorfuTable.class, true, SOME);
+        checkPointAll(getDefaultRuntime());
 
-        map1.put("k1", "v1");
-        map1.put("k2", "v2");
-        map2.put("k3", "v3");
-        map2.put("k4", "v4");
-        map3.put("k5", "v5");
-        map3.put("k6", "v6");
-
-        MultiCheckpointWriter mcw = new MultiCheckpointWriter();
-        mcw.addMap((SMRMap) map1);
-        mcw.addMap((SMRMap) map2);
-        mcw.addMap((SMRMap) map3);
-        mcw.appendCheckpoints(rt1, "author");
-
-        // Create a new runtime with fastloader
-        CorfuRuntime rt2 = new CorfuRuntime(getDefaultConfigurationString())
-                .setLoadSmrMapsAtConnect(true)
-                .connect();
+        CorfuRuntime rt2 = Helpers.createNewRuntimeWithFastLoader(getDefaultConfigurationString());
 
-        assertThatMapIsBuilt(rt1, rt2, "Map1", map1);
-        assertThatMapIsBuilt(rt1, rt2, "Map2", map2);
-        assertThatMapIsBuilt(rt1, rt2, "Map3", map3);
-        assertThatObjectCacheIsTheSameSize(rt1, rt2);
+        assertThatMapsAreBuilt(rt2);
+        assertThatObjectCacheIsTheSameSize(getDefaultRuntime(), rt2);
     }
 
     @Test
     public void canReadWhenTheUserHasNoCheckpoint() throws Exception {
-        CorfuRuntime rt1 = getDefaultRuntime();
-        Map<String, String> map1 = createMap("Map1", rt1);
-        Map<String, String> map2 = createMap("Map2", rt1);
-        Map<String, String> map3 = createMap("Map3", rt1);
-
-        map1.put("k1", "v1");
-        map1.put("k2", "v2");
-        map2.put("k3", "v3");
-        map2.put("k4", "v4");
-        map3.put("k5", "v5");
-        map3.put("k6", "v6");
+        populateMaps(SOME, getDefaultRuntime(), CorfuTable.class, true, SOME);
 
         // Create a new runtime with fastloader
         CorfuRuntime rt2 = new CorfuRuntime(getDefaultConfigurationString())
                 .connect();
 
         FastObjectLoader fsm = new FastObjectLoader(rt2);
         fsm.setLogHasNoCheckPoint(true);
+        fsm.setDefaultObjectsType(CorfuTable.class);
         fsm.loadMaps();
 
-        assertThatMapIsBuilt(rt1, rt2, "Map1", map1);
-        assertThatMapIsBuilt(rt1, rt2, "Map2", map2);
-        assertThatMapIsBuilt(rt1, rt2, "Map3", map3);
-        assertThatObjectCacheIsTheSameSize(rt1, rt2);
-
+        assertThatMapsAreBuilt(rt2);
+        assertThatObjectCacheIsTheSameSize(getDefaultRuntime(), rt2);
     }
 
     @Test
     public void ignoreStreamForRuntimeButNotStreamTails() throws Exception {
```

Comment:
What happens when you white list and ignore a stream?

Reply:
The expectation is that you should not be doing this. Modes are mutually exclusive, so adding a stream to ignore in the whitelist mode should be a prohibited operation. I added an exception when this happens.